src/Daemon/Daemon.cpp

```diff
@@ -460,6 +466,49 @@ namespace usbguard
   {
   }
 
+  pid_t Daemon::pid_fd = -1;
+
+  void Daemon::daemonize(const std::string &pid_file)
+  {
+      USBGUARD_LOG(Trace) << "Starting daemonization";
+
+      pid_t pid = 0;
+
+      USBGUARD_SYSCALL_THROW("Daemonize", (pid = fork()) < 0);
+      if (pid > 0) {
+        usleep(2500); /* We need to die AFTER pidfile is written to... */
+        exit(EXIT_SUCCESS);
+      }
+
+      /* Now we are forked */
+      USBGUARD_SYSCALL_THROW("Daemonize", setsid() < 0);
+      signal(SIGCHLD, SIG_IGN);
+
+
+      USBGUARD_SYSCALL_THROW("Daemonize", (pid = fork()) < 0);
+      if (pid > 0) {
+        exit(EXIT_SUCCESS);
+      }
+
+      /* Now we are forked 2nd time */
+      umask(0047);  /* no need for world-accessible or executable files */ 
+      chdir("/");
+      const int std_fds[] = {STDIN_FILENO, STDOUT_FILENO, STDERR_FILENO};
+      /* We do not need to close all fds because there is only logging open at this point */
+      for (auto fd : std_fds) {
+        close(fd);
+        fd = open("/dev/null", O_RDWR);
```

Comment:
This won't redirect 0,1,2 to /dev/null unless you use the dup2() syscall. It will simply rewrite a local variable and that's all.